### PR TITLE
fix(ui): Add autocomplete for Teams

### PR DESCRIPTION
**Pull request by mikellykels**
Add autocomplete for Teams using the assigned key in Issue search.

Screen Shot 2020-11-22 at 5 30 00 AM
Screen Shot 2020-11-22 at 5 30 10 AM

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/withIssueTags.tsx [hunk 1]
```diff
@@ -72,17 +89,20 @@ const withIssueTags = <P extends InjectedTagsProps>(
...
     setAssigned() {
-      if (this.state.users && this.state.tags.assigned) {
-        const {tags, users} = this.state;
+      if (this.state.tags.assigned) {
+        const {tags, users, teams} = this.state;
         const usernames: string[] = users.map(getUsername);
-        usernames.unshift('me');
-
+        const teamnames: string[] = teams
+          .filter(getTeamName)
```
Comment (scttcper) [inline src/sentry/static/sentry/app/utils/withIssueTags.tsx:96]: i would inline getTeamName and return isMember instead.
`team => team.isMember`

File: src/sentry/static/sentry/app/utils/withIssueTags.tsx [hunk 2]
```diff
@@ -72,17 +89,20 @@ const withIssueTags = <P extends InjectedTagsProps>(
     },
 
     setAssigned() {
-      if (this.state.users && this.state.tags.assigned) {
-        const {tags, users} = this.state;
+      if (this.state.tags.assigned) {
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/utils/withIssueTags.tsx:92]: do need to check for `this.state.users` and `this.state.teams` now?

**Review by scttcper — APPROVED** [review]
lgtm, just the one feedback

File: src/sentry/static/sentry/app/utils/withIssueTags.tsx (hunk 2 above)
Comment (mikellykels) [reply src/sentry/static/sentry/app/utils/withIssueTags.tsx]: Ah, you're right - removed the check 👍

File: src/sentry/static/sentry/app/utils/withIssueTags.tsx [hunk 3]
```diff
@@ -72,25 +81,26 @@ const withIssueTags = <P extends InjectedTagsProps>(
...
-              ...tags.bookmarks,
-              values: usernames,
-            },
+      const {tags, users, teams} = this.state;
+      const usernames: string[] = users.map(getUsername);
+      const teamnames: string[] = teams
+        .filter(team => team.isMember)
+        .map(team => `#${team.name}`);
+      const allAssigned = usernames.concat(teamnames);
+      allAssigned.unshift('me') && usernames.unshift('me');
```
Comment (scttcper) [inline src/sentry/static/sentry/app/utils/withIssueTags.tsx:90]: nit: this is a little strange to one line

Comment (mikellykels) [reply src/sentry/static/sentry/app/utils/withIssueTags.tsx:90]: @scttcper ah ok cool, would you recommend something like this:

```
const allAssigned = [usernames.concat(teamnames), usernames];
allAssigned.map(assigned => assigned.unshift('me'));
```

Comment (scttcper) [reply src/sentry/static/sentry/app/utils/withIssueTags.tsx:90]: the same unshift you have on two lines is fine. It always does both right?

Comment (mikellykels) [reply src/sentry/static/sentry/app/utils/withIssueTags.tsx:90]: ok cool cool, and yeah it'll always do both

**Pull request merged**